FWCore/Framework/interface/EventProcessor.h

```diff
@@ -265,9 +249,14 @@ namespace edm {
     // init() is used by only by constructors
     void init(std::shared_ptr<ProcessDesc>& processDesc, ServiceToken const& token, serviceregistry::ServiceLegacy);
 
-    bool readNextEventForStream(unsigned int iStreamIndex, LuminosityBlockProcessingStatus& iLumiStatus);
+    void readAndMergeRunEntries(std::shared_ptr<RunProcessingStatus>, WaitingTaskHolder);
+    void readAndMergeLumiEntries(std::shared_ptr<LuminosityBlockProcessingStatus>, WaitingTaskHolder);
+
+    void handleNextItemAfterMergingRunEntries(std::shared_ptr<RunProcessingStatus>, WaitingTaskHolder);
+
+    bool readNextEventForStream(WaitingTaskHolder const&, unsigned int iStreamIndex, LuminosityBlockProcessingStatus&);
```

Comment:
I think the answer is no, because the only thing it uses the WaitingTaskHolder for is to call 
```
if (iTask.taskHasFailed()) {
```
It does not start any asynchronous tasks. It is just checking to see if there was a failure on some other thread.